sentry/src/test/java/io/sentry/SentryEnvelopeItemTest.kt

```diff
@@ -68,21 +67,21 @@ class SentryEnvelopeItemTest {
         file.writeBytes(twoMB)
         val attachment = Attachment(file.absolutePath)
 
-        val item = SentryEnvelopeItem.fromAttachment(mock(), attachment)
+        val item = SentryEnvelopeItem.fromAttachment(attachment)
 
         assertAttachment(attachment, twoMB, item)
     }
 
     @Test
     fun `fromAttachment with non existent file`() {
-        val logger = mock<ILogger>()
         val attachment = Attachment("I don't exist", "file.txt")
 
-        val item = SentryEnvelopeItem.fromAttachment(logger, attachment)
+        val item = SentryEnvelopeItem.fromAttachment(attachment)
 
-        assertAttachment(attachment,
-                "Reading the attachment ${attachment.pathname} failed, because the file located at " +
-                        "the path is not a file.", item)
+        assertFailsWith(SentryEnvelopeException::class, "Reading the attachment ${attachment.pathname} failed, because the file located at " +
+                "the path is not a file.") {
+            item.data
+        }
```

Comment:
M: you can use the reified option

eg:
`assertFailsWith< SentryEnvelopeException > { val item = SentryEnvelopeItem.fromAttachment(attachment) }`

Reply:
Good point thanks @marandaneto 👏🏼